code/part-one/mining.js

```diff
@@ -0,0 +1,95 @@
+'use strict';
+
+const { createHash } = require('crypto');
+const signing = require('./signing');
+const { Block, Blockchain } = require('./blockchain');
+
+
+class MineableTransaction {
+  constructor(privateKey, recipient = null, amount) {
```

Comment:
Having a default parameter `recipient = null` followed by a required argument `amount` seems a little wonky. Should we put make `recipient = null` the last argument or not give it a default value of `null`?

Reply:
The original order of `to`, `from`, `amount` makes a lot of sense to me, just semantically. The `= null` is really just a clue that you can `null` it. You are right it doesn't actually really do anything functional. I like communicating that in the function signature, especially since that is one of the pieces of information students will have. It could be omitted though if it seems too odd.